gradle/lib/dependabot/gradle/file_parser.rb

```diff
@@ -238,7 +238,7 @@ def plugin_dependencies(buildfile)
       end
 
       def format_plugin_version(version)
-        version&.match?(/^\w+$/) ? "$#{version}" : version
+        version&.match?(/^['"].*['"]$/) ? version[1..-2] : "$#{version}"
```

Comment:
Of course!

So, there's two cases for defining plugins here:

```kotlin
// A version literal
id("com.google.protobuf") version "0.8.4" apply false

// Something dynamic
val helmVersion = "1.6.0"
id("org.unbroken-dome.helm") version helmVersion apply false
```

Those lead to different code paths because in the literal case we'll leave the literal version alone, while in the dynamic property case we'll try to find a value for the properlty.

Previously we would parse what's between the `version` and `apply` with the following regex:

https://github.com/dependabot/dependabot-core/blame/e5bb0997b216cd7bf70b12f921318442804a3036/gradle/lib/dependabot/gradle/file_parser.rb#L227.

Note that the regex would match quoted literals, but not include the quotes themselves in the matched part.

Now we're dealing with something like:

```kotlin
id("com.google.protobuf") version libs.versions.protoPlugin apply false
```

The problem is that this was being misidentified as a version literal, because `libs.versions.protoPlugin` does not match `/^\w+$/` because of the dots (`.`).

So my approach was to change the regex above so that the literals are actually included in the named match, and check whether the matched part is quoted or not. But to keep the rest of the logic as it was before, I need to discard the matched quotes once I have discriminated the two cases. That's what `version[1..-2]` does, discard the first and the last characters from the string, which if the string is quoted, essentially means "unquoting" it.

Hopefully this makes sense!

Reply:
That makes sense, thanks!

Normally I would do something like `"some.version.reference".delete('"')` since it's more explicit, but I think this works fine 👍🏾

Reply:
You're totally right about that being more explicit, although we would also need to remove `'` too. How about extracting an `unquote(string)` method, I think that would make the code more clear.

Reply:
Sure, that sounds like a good middle ground!

Reply:
Done @Nishnha! I'll merge this!